src/event.lib.php

```diff
@@ -766,14 +771,16 @@ public static function setNewPassword($user_id = 0)
      * @since 1.8.25
      *
      * @param WP_User $user WP_User object.
-     * @return string|false Returns a password reset key as a string, false otherwise.
+     * @return string|WP_Error Returns a password reset key as a string, WP_Error otherwise.
      */
     public static function GetPasswordResetKey($user)
     {
         global $wp_hasher;
 
+        $key_error = new WP_Error( 'no_password_reset' );
+
         if (!($user instanceof WP_User)) {
-            return false;
+            return $key_error;
```

Comment:
The [`get_password_reset_key`](https://core.trac.wordpress.org/browser/tags/5.6/src/wp-includes/user.php#L2485) function returns an error with different code. I do not know how these errors will be handled, but maybe it makes sense this method to return the same errors. Thoughts?

Reply:
I don't think we're going to be checking why we failed to generate the `$key`, but if you feel like we should maintain an implementation as close as `get_password_reset_key` as possible, I see no problem. I just thought it was not necessary for our use case. Please let me know what you think! :)

Reply:
If showing different error messages improves the UX, we could use separate codes/messages, otherwise it is fine as is.